text/0000-builder-spec.md

```diff
@@ -0,0 +1,270 @@
+# Meta
+[meta]: #meta
+- Name: Builder Spec
+- Start Date: 2020-09-11
+- Author(s): [@dfreilich](https://github.com/dfreilich)
+- RFC Pull Request: (leave blank)
+- CNB Pull Request: (leave blank)
+- CNB Issue: (leave blank)
+- Supersedes: N/A
+
+# Summary
+[summary]: #summary
+A Builder is a critical tool in the CNB ecosystem, and yet, is largely [unspecified][spec-issue-builder]. This RFC proposes that we should add a rigorous description of builders in the [distribution specification][distribution-spec], and clarify the usage/existence of builders in the [platform specification][platform-spec]
+
+# Definitions
+[definitions]: #definitions
+* **[Builder][builder-docs]** &rarr; A **builder** is an image that bundles all the bits and information on how to build your apps, such as buildpacks and build-time image, as well as executes the buildpacks against your app source code.
+* **[Buildpack][buildpacks-docs]** &rarr; A **buildpack** is a unit of work that inspects your app source code and formulates a plan to build and run your application.
+* **[Platform][platform-docs]** &rarr; A **platform** uses a lifecycle, buildpacks (packaged in a builder), and application source code to produce an OCI image.
+* **[Stack][stack-docs]** &rarr; A stack provides the buildpack lifecycle with build-time and run-time environments in the form of images.
+
+# Motivation
+[motivation]: #motivation
+Builders are a vital part of the CNB ecosystem as it currently stands; they are used to distribute the lifecycle and buildpacks to every platform, including:
+* pack
+* Tekton
+* pack-orb
+* kpack
+etc.
+
+However, at this point, builders are underspecified, and a rigorous technical specification would fill a needed hole in the current documentation.
+
+# What it is
+[what-it-is]: #what-it-is
+A Builder consists of four items:
```

Comment:
I think the stack information including a reference to the `run-image` and potentially `run-image-mirrors` should be called out separately here.

Reply:
done in 41b9a4ae471e3490de686f66e81fa8bbd999a262